pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -0,0 +1,207 @@
+			desiredControllerManagerArgumentVals[nodeMonitorGracePeriodArgument] = strconv.Itoa(int(apiconfigv1.MediumNodeMonitorGracePeriod))
+		case apiconfigv1.LowUpdateSlowReaction:
+			desiredControllerManagerArgumentVals[nodeMonitorGracePeriodArgument] = strconv.Itoa(int(apiconfigv1.LowNodeMonitorGracePeriod))
+		}
+
+		condition := metav1.Condition{
+			Type:   apiconfigv1.KubeControllerManagerDegraded,
+			Status: metav1.ConditionUnknown,
+		}
+		// do nothing if workerLatencyProfile is not set
+		if configNodeObj.Spec.WorkerLatencyProfile == "" {
+			// TODO: in case worker latency profile is transitioned
+			// from Default/Medium/Low to ""
+			// check that the required extendedArgs should vanish
+
+			condition.Reason = reasonWorkerLatencyProfileEmpty
+			condition.Message = "worker latency profile not set on cluster"
+			_, err := c.updateConfigNodeStatus(condition)
+			return err
+		}
+
+		// For each revision, check that the configmap for that revision have
+		// correct argument values or not
+		revisionsHaveSynced := true
+		for _, revision := range uniqueRevisions {
+			configMapNameWithRevision := fmt.Sprintf("%s-%d", controllerManagerConfigMapName, revision)
+			configMap, err := c.configMapClient.ConfigMaps(operatorclient.TargetNamespace).Get(ctx, configMapNameWithRevision, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			matches := true
+			for arg, val := range desiredControllerManagerArgumentVals {
+				match, err := configMatchesControllerManagerArgument(configMap, arg, val)
+				if err != nil {
+					return err
+				}
+				matches = matches && match
+			}
+			if !matches {
+				revisionsHaveSynced = false
+				break
+			}
+		}
+
+		condition.Status = metav1.ConditionTrue
+		condition.Reason = reasonWorkerLatencyProfileUpdated
+		if revisionsHaveSynced {
+			condition.Type = apiconfigv1.KubeControllerManagerComplete
+			condition.Message = "all kube-controller-manager(s) have updated latency profile"
+		} else {
+			condition.Type = apiconfigv1.KubeControllerManagerProgressing
+			condition.Message = "kube-controller-manager(s) are updating the latency profile"
+		}
+		_, err = c.updateConfigNodeStatus(condition)
+		return err
+	}()
+
+	condition := operatorv1.OperatorCondition{
+		Type:   "WorkerLatencyControllerDegraded",
+		Status: operatorv1.ConditionFalse,
+	}
+	if syncErr != nil {
+		condition.Status = operatorv1.ConditionTrue
+		condition.Reason = "Error"
+		condition.Message = syncErr.Error()
+	}
+	if _, _, updateErr := v1helpers.UpdateStatus(ctx, c.operatorClient, v1helpers.UpdateConditionFn(condition)); updateErr != nil {
+		return updateErr
+	}
+	return syncErr
+}
+
+// configHasControllerManagerArgument checks if the specified config map containing kcm node config
+// contains the specified argument and value in observedconfig.extendedarguments field
+func configMatchesControllerManagerArgument(configMap *corev1.ConfigMap, arg, value string) (bool, error) {
```

Comment:
could we have a simple unit test for this function?

Reply:
Added. 

Also, added unit tests for the config observer.